textile/features.textile

```diff
@@ -688,7 +688,9 @@ h3(#realtime-channel). RealtimeChannel
 ** @(RTL7b)@ Subscribe with a name argument and a listener argument subscribes a listener to only messages whose @name@ member matches the string name
 ** @(RTL7c)@ This clause has been replaced by "@RTL7g@":#RTL7g. It was valid up to and including specification version @3.0@.
 ** @(RTL7g)@ If the @attachOnSubscribe@ channel option is @true@, implicitly attaches the @RealtimeChannel@ if the channel is in the @INITIALIZED@, @DETACHING@, or @DETACHED@ states. The optional callback, if provided, is called according to "@RTL4d@":#RTL4d based on the implicit attach operation. The listener will always be registered regardless of the implicit attach result
-** @(RTL7h)@ If the @attachOnSubscribe@ channel option is @false@, the optional callback must not be called.
+** @(RTL7h)@ If the @attachOnSubscribe@ channel option is @false@, then the behaviour depends on the public API that a given SDK uses to communicate the result of an "@RTL7g@":#RTL7g implicit attach:
+*** If the SDK’s API accepts an optional callback (or equivalent, for example it provides two overloads of @#subscribe@, one which accepts such a callback and one which doesn’t) to communicate the result of an "@RTL7g@":#RTL7g implicit attach, then it is a programmer error to provide such a callback when the @attachOnSubscribe@ channel option is @false@. This programmer error should be handled in an idiomatic fashion (e.g. by halting the program).
+*** If the SDK’s API accepts a non-optional callback (or equivalent, for example it returns a @ChannelStateChange?@) to communicate the result of an "@RTL7g@":#RTL7g implicit attach, then when the @attachOnSubscribe@ channel option is @false@ this callback should be called with no argument.
```

Comment:
Actually, `If the SDK’s API accepts a non-optional callback` feels confusing. 
Because, if we are applying this case for ably-go, `ably-go` doesn't accept any optional/non-optional callback for subscribe.

Reply:
Rather, we can say, if result of subscribe operation is communicated as a async/sync blocking operation that includes result of implicit attach, then null error should be returned for the same.

Reply:
Also, in case of `ably-go`, it returns `unsubscribe` method that removed listener instead of returning 
`ChannelStateChange`

Reply:
I’ve removed the "accepts a non-optional callback" language and instead reworded it as these two cases:

- it accepts an optional callback
- anything else